### DISCUSSION TITLE
Tad 0.10.0: A cross-platform tabular data viewer desktop app, powered by DuckDb

### DISCUSSION THREAD

**Discussion by antonycourtney** [duckdb/duckdb#3566] [Show and tell]
I recently released an update to [Tad](https://www.tadviewer.com) ([github](https://github.com/antonycourtney/tad)), a fast, free cross-platform tabular data viewer application, now powered by DuckDb!

Tad is a low friction way to explore tabular data files (CSV and Parquet files). A "dirty little secret" of the data world is that our workflows often involve using commercial spreadsheet apps like Excel to take a quick peek at the contents of a CSV file.  Tad provides a pleasant (and free) alternative to this, and (thanks to DuckDb!) is much faster than Excel at opening CSV files.  How much faster?  On my 2019 MacBook Pro, opening MetObjects.csv, a 230 MB CSV with 450k rows takes around 33 seconds to open with the latest version of Excel. This same file opens in under 5 seconds in Tad.  

Tad also supports Parquet files. Since DuckDb operates on Parquet files in place without an extra import step, the speed for opening Parquet files is pretty mind-blowing.  Here is Tad browsing a directory of multi-million-row Parquet files from the TPC-H-SF10 benchmark data set:

![Tad-parquet](https://user-images.githubusercontent.com/4683413/166626636-2cebbcd7-855c-487d-8c29-bf577885d536.gif)

For data exploration, the core of Tad is a  a hierarchical pivot table that allows you to specify a combination of pivot, filter, aggregate, sort, column selection, column ordering and basic column formatting operations directly in the UI. Tad delegates to a SQL database for storage and analytics, and generates SQL queries to perform all analytic operations specified in the UI.  Here is a quick view of using Tad to pivot and sort a tabular data file:

![tad-metobjects-pivoted](https://user-images.githubusercontent.com/4683413/166625007-199dccd5-da04-449a-9bf0-236da647cd74.png)

Of particular interest to DuckDb users, Tad can also serve as a lightweight browser for DuckDb and SQLite database files.  Just run:
```sh
$ tad myDatabase.duckdb
```
from the command line to get a browsable view of the tables in a DuckDb file.  (Tad was built with DuckDb 0.3.3, so will only be able to open db files saved with that version of DuckDb for now.)

You can download pre-built installers for Mac, Linux and Windows from the [Tad github releases page](https://github.com/antonycourtney/tad/releases/latest).  

I hope DuckDb users find Tad a useful tool.  [Feedback welcome!](mailto:tad-feedback@tadviewer.com)

**Comment by Alex-Monahan** [comment]
It certainly looks really helpful! Thanks for building it and for sharing!

You are also welcome to post a How To Guide in the DuckDB docs repo if you would like. Here is an example of the couple of files to change:
https://github.com/duckdb/duckdb-web/pull/192
